akka-stream/src/main/scala/akka/stream/impl/fusing/Interpreter.scala

```diff
@@ -662,16 +689,26 @@ private[akka] class OneBoundedInterpreter(ops: Seq[Stage[_, _]],
         case a: AsyncStage[Any, Any, Any] @unchecked ⇒
           a.context = new EntryState("async", op)
           activeOpIndex = op
-          a.initAsyncInput(a.context) // TODO remove asyncInput? it's like preStart
-
-        case _ ⇒
+          a.preStart(a.context)
+
+        case a: AbstractStage[Any, Any, Any, Any, Any] @unchecked ⇒
+          val state = new EntryState("stage", op)
+          a.context = state
+          try a.preStart(state) catch {
+            case NonFatal(ex) ⇒
+              failures ::= InitializationFailure(op, ex) // not logging here as 'most downstream' exception will be signalled via onError
+            // TODO could use decider here, but semantics become a bit iffy (Resume => ignore error in prestart? Doesn't sound like a good idea).
+          }
       }
       op += 1
     }
+    failures
   }
 
   private def finishCurrentOp(): Unit = {
-    pipeline(activeOpIndex) = Finished.asInstanceOf[UntypedOp]
+    try pipeline(activeOpIndex).postStop()
```

Comment:
add a test case for pushAndFinish ops. Look at the current test cases for pushAndFinish in InterpreterSpec and adapt them. The reason is because pushAndFinish uses a PushFinished marker and I want to be sure that all cases are covered.